Dockerfile

```
@@ -1,6 +1,11 @@
-FROM golang:alpine3.15 as build
+FROM golang:1.19-alpine3.16 as build
 RUN apk add libc6-compat gcc musl-dev
 WORKDIR /build/
+
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
+
 COPY . .
 RUN --mount=type=cache,target=/go/pkg/mod go mod download
 RUN --mount=type=cache,target=/root/.cache/go-build go build -o /build/poet
```

A second `go mod download` is probably not needed and we have a Makefile target for `build`.

```suggestion
RUN --mount=type=cache,id=build,target=/root/.cache/go-build make build
```